test: bsim: Fix missed bsim warnings

alxelax opened the pull request:
PR fixes https://github.com/zephyrproject-rtos/zephyr/issues/38303
It adds a lost compiler option to prohibit warnings
It fixes warnings in BabbleSim hw repo. (need to be merged first.)

alxelax committed:
bsim: test: fix missing warnings in bsim test build system

bsim test build system hides compile warnings.
The problem had been fixed before but due to a regression came again.
This commit adds extra cc flags to prevent this.

Signed-off-by: Aleksandr Khromykh <aleksandr.khromykh@nordicsemi.no>

alxelax committed:
test: bsim: set bsim hw head on fixed version

Set bsim hw repo head on the version with fixed warnings

Signed-off-by: Aleksandr Khromykh <aleksandr.khromykh@nordicsemi.no>

aescolar reviewed (changes requested):
The PR is fine (note that you have a couple of typos in the commits messages*, that you may as well fix when updating the west reference)
By now just blocking it to avoid a merge while we are pointing to a PR (anybody, feel free to dismiss this review once the reference is updated)

bsim: test: fix missing -of- warnings [..]

bsim test build system hides compile warnings-.-
The problem ha-s-+d+ been fixed before bu+t due to a+ regression came again.
This commit adds extra cc flags to prevent this.

alxelax commented:
> The PR is fine (note that you have a couple of typos in the commits messages*, that you may as well fix when updating the west reference) By now just blocking it to avoid a merge while we are pointing to a PR (anybody, feel free to dismiss this review once the reference is updated)
> 
> bsim: test: fix missing -of- warnings [..]
> 
> bsim test build system hides compile warnings-.- The problem ha-s-+d+ been fixed before bu+t due to a+ regression came again. This commit adds extra cc flags to prevent this.

fixed commit message and added real hash as a reference
